pygments/lexers/berry.py

```diff
@@ -0,0 +1,99 @@
+    
+    name = 'Berry'
+    aliases = ['berry', 'be']
+    filenames = ['*.be']
+    mimetypes = ['text/x-berry', 'application/x-berry']
+
+    _name = r'\b[^\W\d]\w*'
+
+    # (def)\b((?:\s)*)
+    # (def)\b(\s*$)(?<!\()
+    # (def)\b((?:\s)*)(?=[^\W\d]\w*)
+
+    tokens = {
+        'root': [
+            include('whitespace'),
+            include('numbers'),
+            include('keywords'),
+            (r'(def)\b((?:\s)*)' + '(?=' + _name + ')', bygroups(Keyword.Declaration, Text), 'funcname'),
+            (r'(class)\b((?:\s)*)', bygroups(Keyword.Declaration, Text), 'classname'),
+            (r'(import)\b((?:\s)*)', bygroups(Keyword.Namespace, Text), 'import'),
+            include('expr')
+        ],
+        'expr': [
+            (r'[^\S\n]+', Text),
+            (r'\.\.|[~!%^&*+=|?:<>/-]', Operator),
+            (r'[(){}\[\],.;]', Punctuation),
+            include('controls'),
+            include('builtins'),
+            include('funccall'),
+            include('member'),
+            include('name'),
+            include('strings')
+        ],
+        'whitespace': [
+            (r'\s+', Whitespace),
+            (r'#-(.|\n)*?-#', Comment.Multiline),
+            (r'#(\n|[\w\W]*?\n)', Comment.Single)
+        ],
+        'keywords': [
+            (words((
+                'as', 'break', 'continue', 'import', 'static', 'self', 'super'), 
+                suffix=r'\b'), Keyword.Reserved),
+            (r'(true|false|nil)\b', Keyword.Constant),
+            (r'(var|def)\b', Keyword.Declaration)
+        ],
+        'controls': [
+            (words((
+                'if', 'elif', 'else', 'for', 'while', 'do', 'end', 'break', 
+                'continue', 'return', 'try', 'except', 'raise'),
+                suffix=r'\b'), Operator.Word)
+        ],
+        'builtins': [
+            (words((
+                'assert', 'bool', 'input', 'classname', 'classof', 'number', 'real',
+                'bytes', 'compile', 'map', 'list', 'int', 'isinstance', 'print',
+                'range', 'str', 'super', 'module', 'size', 'issubclass', 'open',
+                'file', 'type', 'call'), 
+                suffix=r'\b'), Name.Builtin)
+        ],
+        'numbers': [
+            (r'0[xX](?:_?[a-fA-F0-9])+', Number.Hex),
+            (r'[-]?\d+', Number.Integer),
+            (r'([-]?\d*\.?\d+)(?:[eE]([-+]?\d+))?', Number.Float)
+        ],
+        'name': [
+            (_name, Name)
+        ],
+        'funcname': [
+            (_name, Name.Function, '#pop')
+        ],
+        'funccall': [
+            ( _name + r'(?=\s*\()', Name.Function, '#pop')
+        ],
+        'member': [
+            ( r'(?<=\.)' + _name + r'\b(?!\()', Name.Attribute, '#pop')
+        ],
+        'classname': [
+            (_name, Name.Class, '#pop'),
+        ],
+        'import': [
+            (r'(\s+)(as)(\s+)', bygroups(Text, Keyword, Text)),
+        ],
+        'strings': [
+            (r'"(?:\\?.)*?"', String.Double, '#pop'),
```

Comment:
The pattern `"(\\?.)*?"` suffers from catastrophic backtracking. Try lexing

```
"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
```

(without a quote at the end). It will take a very long time. The problem is that you can have a backslash anywhere but at the end of the string match in two ways, either as `\\.` or just `.`. Thus, the recognition has exponential complexity if the regexp engine cannot find a match at the end. Please change this to `"([^\\]|\\.)*?"`.